ops/framework.py

```diff
@@ -199,6 +199,11 @@ def __init__(self, parent, key):
         else:
             self.framework = parent.framework
             self.handle = Handle(parent, kind, key)
+        if parent is not self:
+            # Framework.__init__ passes parent=self, and we don't want to track ourself (also framework._objects may not be initialized yet)
+            if self.handle.path in self.framework._objects:
+                raise RuntimeError(f"two objects claiming to be {self.handle.path} have been created")
+            self.framework._objects[self.handle.path] = self
```

Comment:
Instead of having that logic here, it would probably be nicer to have something like this:
```
self.framework._track(self)
```
Then, inside `_track`, we can exclude the framework in a more obvious way:
```
if obj is self:
    return
```
This will make sense right next to `_forget`.